src/components/App/index.tsx

```diff
@@ -61,8 +80,38 @@ export class AppBase extends React.Component<Props> {
   static defaultProps = {
     _fetchCurrentUser: fetchCurrentUser,
     _log: log,
+    _tracking: tracking,
   };
 
+  resourceObserver: PerformanceObserver | MockObserver;
+
+  createResourceObserver() {
+    const { _mockObserver, _tracking } = this.props;
+
+    if (_mockObserver) {
+      return _mockObserver;
+    }
+
+    const observer = new PerformanceObserver((list) => {
```

Comment:
So, I know you mentioned this in a PR comment and I am not super familiar with how mocking such web APIs is done elsewhere already, but I'm not a fan of the mocking approach here. It feels like a weird compromise: you have some test/mocking specific code in `createResourceObserver` (creating a huge difference in how the whole thing behaves in tests and in "production") but at the same time don't go all the way and just return the mock directly, bypassing a bunch of code.

What's preventing you from instantiating `resourceObserver` here ?

Reply:
I too am not very happy with this approach, but I've been trying to figure out how to get the whole thing mocked and thus far have had no luck. I suppose if I can create a true mocked version of `PerformanceObserver` I should be able to instantiate it here. I'll give that a try. It's fine if this doesn't make it into the tag. I'd rather get this mocking working fully.

Reply:
I have made an improvement to the mocking, but it's still unable to fully test the observer. I injected a mock class so that we can better test the constructor, and we do not have separate paths for test vs. production, which was poor.

I am still unable to figure out how to test the body of the observer. The callback is called whenever a performance entry for a `resource` is recorded, but I can't figure out how to emulate that via a mock and a test. I will continue searching, and will see if I can ask around Mozilla, but at this point I'm just not sure how to do it.

I think what is here is an improvement over the previous version, and it should give us a bit more coverage, but coverage for the actual observer callback is still missing.